src/sentry/static/sentry/app/views/eventsV2/miniGraph.tsx

```diff
@@ -34,8 +34,13 @@ class MiniGraph extends React.Component<Props> {
   }
 
   render() {
-    const {organization, api, selection, query} = this.props;
-    const {start, end, period} = selection.datetime;
+    const {organization, api, location, eventView} = this.props;
+
+    const apiPayload = eventView.getEventsAPIPayload(location);
+    const query = apiPayload.query;
+    const start = getUtcToLocalDateObject(apiPayload.start);
+    const end = getUtcToLocalDateObject(apiPayload.end);
+    const period: string | undefined = apiPayload.statsPeriod as any;
```

Comment:
Do you need the type def if you are casting to any?

Reply:
The type for `apiPayload.statsPeriod` is detected as `string | string[] | null | undefined`. But it's guaranteed that `statsPeriod ` will be  `string | undefined`. Hence why I'm casting it like this.